overlord/snapstate/autorefresh.go

```diff
@@ -74,12 +79,51 @@ func (m *autoRefresh) NextRefresh() time.Time {
 
 // LastRefresh returns when the last refresh happened.
 func (m *autoRefresh) LastRefresh() (time.Time, error) {
-	var lastRefresh time.Time
-	err := m.state.Get("last-refresh", &lastRefresh)
-	if err != nil && err != state.ErrNoState {
+	return getTime(m.state, "last-refresh")
+}
+
+// EffectiveRefreshHold returns the time until to which refreshes are
+// held if refresh.hold configuration is set and accounting for the
+// max postponement since the last refresh.
+func (m *autoRefresh) EffectiveRefreshHold() (time.Time, error) {
+	var holdTime time.Time
+
+	tr := config.NewTransaction(m.state)
+	err := tr.Get("core", "refresh.hold", &holdTime)
+	if err != nil && !config.IsNoOption(err) {
```

Comment:
`tr.GetMaybe` would save you the `IsNoOption` boilerplate (granted it's a small boiler, or a small plate)

Reply:
there's a couple more places that could use this change, I will do a separate PR